contracts/base/FallbackManager.sol

```diff
@@ -0,0 +1,57 @@
+pragma solidity ^0.5.0;
+
+import "../common/SelfAuthorized.sol";
+
+/// @title Fallback Manager - A contract that manages fallback calls made to this contract
+/// @author Richard Meissner - <richard@gnosis.pm>
+contract FallbackManager is SelfAuthorized {
+
+    // TO BE DISCUSSED: Using this with assembly storage access costs about 100 gas more for a read, and 50 for a write,
```

Comment:
Would love your opinions on that @denisgranha @cag @Georgi87 @Uxio0

Reply:
The only thing is bugging me is that for the `FallbackManager` variables are stored using hash (I agree is very useful to prevent messing the proxy storage) but for example for the `ModuleManager` we are depending on having the variables sorted. Is that for compatibility with old version of Safes?

Reply:
So this doesn't interfer with any variable ordering, but if we would use a variable I used the correct order of inheritance in the GnosisSafe contract.

Was that your question?

Reply:
My question is that, shouldn't we use the same storage logic for the rest of the modules?

Reply:
I was thinking that using this in newer code, minimizes the risk of interfering with existing code

Reply:
Agree :+1:

Reply:
I agree it is a good practice to ensure there is no variable overwriting and worth the additional gas costs. This should probably be documented in our docs as best practice in case someone wants to extend the Safe functionality.

Reply:
I added a guidlines.md to document these specifics